plugins/serendipity_plugin_superuser/serendipity_plugin_superuser.php

```diff
@@ -46,7 +46,8 @@ function generate_content(&$title)
             $base = $serendipity['serendipityHTTPPath'];
         }
 
-        $link = $base . ($serendipity['rewrite'] == 'none' ? $serendipity['indexFile'] .'?/' : '') . PATH_ADMIN;
+        //$link = $base . ($serendipity['rewrite'] == 'none' ? $serendipity['indexFile'] .'?/' : '') . PATH_ADMIN;
+        $link = $base . 'serendipity_admin.php';
```

Comment:
Why did you change this? In rewrite environments it would be nice to use a short-hand established link like /admin?

Reply:
My debugger monitored the initialization of the serendipity framework and I found out that by clicking on the 'login' link, the call was first to `/index.php/?admin` only to get rerouted to `/serendipity_admin.php` which did all the initialization again which was also part of the lang setting problems. Because the code primer said that all the routing for the backend goes via the /serendipity_admin.php it seemed logically that the link should point there. Sorry if I did not get the concept of the rewrite engine, but in the end, the link should not point at index.php, according to the documentation.

Reply:
OK in that case, yeah - for non-RewriteEngine rewrites it does mean a separate reroute. But when URL rewriting is enabled, it directly maps to the right URL. It wasn't a good design decision to prefix many files with "serendipity_", so where possible I'd like to keep the URLs more clean :D. 

So the best variant should be to use serendipity_admin.php when URL rewriting is off:
```
$link = $base . ($serendipity['rewrite'] == 'none' ? 'serendipity_admin.php' : PATH_ADMIN); ```

that could work?